### ISSUE TITLE
exporter doesn't send the username to the devices

### ISSUE THREAD

**Issue by StefanescuCristian** [oGi4i/mikrotik-exporter#10]
Hi,

I don't know if it's my setup or not, I've recreated the docker container and removed and added the user again on mikrotik and I'm lost.

It seems that the exporter doesn't send the username (or at least that is what I see in the MK device logs) and the exporter cannot login:

![image](https://user-images.githubusercontent.com/1718660/161633174-ecf3704b-0426-4f92-ae03-ef12e72b4879.png)

ROS 7.1.5, tried on arm and mipsbe, tried with and without user-manager package installed.

My config.yml is something like this:

```
devices:
  - name: MK-Berceni
    address: <redacted>
    user: prometheus
    password: <redacted>
  - name: MK-SW-Berceni
    address: <redacted>
    user: prometheus
    password: <redacted>
  - name: MK-GW-Bragadiru
    address: <redacted>
    user: prometheus
    password: <redacted>

features:
  bgp: false
  dhcp: true
  dhcpv6: true
  dhcpl: true
  dhcp_leases: true
  routes: true
  pools: true
  optics: true
  wlansta: true
  wlanif: true
  ipsec: true
  ipsec-peers: true
  monitor: true
  health: true
  conntrack: true
  pool: true
  resource: true
```

```
docker run --rm  -p 9543:9436 -v $PWD:/config ogi4i/mikrotik-exporter -config-file /config/mkt-config.yml
{"device":"MK-Berceni","error":"failed to create client: from RouterOS device: invalid user name or password (6)","level":"error","msg":"failed to collect metrics","time":"2022-04-04T21:10:21Z"}
{"device":"MK-GW-Bragadiru","error":"failed to create client: from RouterOS device: invalid user name or password (6)","level":"error","msg":"failed to collect metrics","time":"2022-04-04T21:10:21Z"}
{"device":"MK-SW-Berceni","error":"failed to create client: from RouterOS device: invalid user name or password (6)","level":"error","msg":"failed to collect metrics","time":"2022-04-04T21:10:21Z"}
```

Any ideas?

**Comment by dumbasPL** [comment]
Same here. ROS 7.2 arm
Was working before just before.

Here is the log from watchtower
```
time="2022-03-28T06:00:55+02:00" level=info msg="Found new ogi4i/mikrotik-exporter:latest image (bf26ec3ea2b9)"
time="2022-03-28T06:00:56+02:00" level=info msg="Stopping /mikrotik-exporter (4660cdd29a6d) with SIGTERM"
time="2022-03-28T06:01:05+02:00" level=info msg="Creating /mikrotik-exporter"
time="2022-03-29T06:00:08+02:00" level=info msg="Found new ogi4i/mikrotik-exporter:latest image (9d2df581223d)"
time="2022-03-29T06:01:14+02:00" level=info msg="Stopping /mikrotik-exporter (5c807562fc49) with SIGTERM"
time="2022-03-29T06:01:15+02:00" level=info msg="Creating /mikrotik-exporter"
```

The last logs I have in Prometheus are from 2022-03-28 so it broke after the update on 2022-03-27

for some reason, older versions are not available on the docker hub. @oGi4i mid tagging and uploading the old image to docker hub. The one that was before v1.1.0 so we can have a working exporter until this gets fixed.

**Comment by dumbasPL** [comment]
Ok found it.  
TLDR: `user` was chnaged to `username` in 11621f0f97b8eb0d6872497f845f6e77085e4f4d  
So all you have to do to fix it is to change `user` to `username` in your config

The readme was updated but a changelog would be nice to have. Especially for a breaking change like this that break all existing configs

**Comment by StefanescuCristian** [comment]
@dumbasPL I can confirm that changing user to username in config did the job to fix this.

**Comment by oGi4i** [comment]
Sorry for that, guys. I accidently broke some key features during refactoring of the project and the latest image got overwritten.
Please refer to the README.md for the current config example.

Also try not to use the latest image in the future, because from now on I'll be issuing tagged semver images on updates and the latest image will always be same as the latest tagged image. So using a fixed tagged image will guarantee nothing will break on update.

**Comment by dumbasPL** [comment]
> Also try not to use the latest image in the future

Some form of a changelog would still be nice. Maybe use the github releases page. Just create a release from a tag and note everything that changed. Or maybe even a simple `CHANGELOG.md` would be nice to have.

**Comment by oGi4i** [comment]
Thanks for the feedback.
I'll try to include changes on the Release page.
